### PR TITLE
Upgrade to Pulumi 3.9.1 and Pulumi-terraform-bridge v3.5.0

### PR DESCRIPTION
**Pull request by stack72**

### CODE REVIEW

**Review by justinvp — COMMENTED** [review]
This PR needs the `Makefile` change like https://github.com/pulumi/pulumi-kubernetes/pull/1664

**Comment by stack72** [comment]
> This PR needs the `Makefile` change like [pulumi/pulumi-kubernetes#1664](https://github.com/pulumi/pulumi-kubernetes/pull/1664)

great catch! fixed

File: sdk/dotnet/S3/Inputs/BucketWebsiteArgs.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public sealed class BucketWebsiteArgs : Pulumi.ResourceArgs
         /// describing redirect behavior and when redirects are applied.
         /// </summary>
         [Input("routingRules")]
-        public InputUnion<string, ImmutableArray<Input<string>>>? RoutingRules { get; set; }
+        public InputUnion<string, ImmutableArray<string>>? RoutingRules { get; set; }
```
Comment (justinvp) [inline sdk/dotnet/S3/Inputs/BucketWebsiteArgs.cs:38]: @pgavlin, this looks suspicious, is this expected?

Comment (stack72) [reply sdk/dotnet/S3/Inputs/BucketWebsiteArgs.cs:38]: yeah this is indeed breaking I believe

Comment (pgavlin) [reply sdk/dotnet/S3/Inputs/BucketWebsiteArgs.cs:38]: Technically this is unbreaking an earlier break. I’ll find the p/p PR for context.

Comment (komalali) [reply sdk/dotnet/S3/Inputs/BucketWebsiteArgs.cs:38]: I think this is the PR @pgavlin is referring to https://github.com/pulumi/pulumi/pull/7601

**Review by justinvp — APPROVED** [review]
LGTM

**Pull request merged**